libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2634,6 +2693,34 @@ map<ContractDefinition const*, vector<ASTPointer<frontend::Expression>>> SMTEnco
 	return baseArgs;
 }
 
+set<FunctionCall const*> SMTEncoder::collectABICalls(ASTNode const* _node)
+{
+	struct ABIFunctions: public ASTConstVisitor
+	{
+		ABIFunctions(ASTNode const* _node) { _node->accept(*this); }
+		void endVisit(FunctionCall const& _funCall)
+		{
+			static set<FunctionType::Kind> const abiKinds{
+				FunctionType::Kind::ABIEncode,
+				FunctionType::Kind::ABIEncodePacked,
+				FunctionType::Kind::ABIEncodeWithSelector,
+				FunctionType::Kind::ABIEncodeWithSignature,
+				FunctionType::Kind::ABIDecode
+			};
+			if (*_funCall.annotation().kind == FunctionCallKind::FunctionCall)
+			{
+				auto const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
+				if (abiKinds.count(funType.kind()))
+					abiCalls.insert(&_funCall);
```

Comment:
Cosmetics only: I'd have gone for a ``switch`` instead of a ``set`` lookup here.

Reply:
Changed